### ISSUE TITLE
Test suite fails if PWD is a symlinked directory

### ISSUE THREAD

**Issue by gjasny** [ccache/ccache#916]
While working on the homebrew ccache packag I also wanted to run the ccache test suite after compiling. But two tests always fail:

```
The following tests FAILED:
	  3 - test.basedir (Failed)
	 21 - test.nocpp2 (Failed)
```

### How to reproduce ###

1. Build ccache on MacOS
2. `mkdir /tmp/testcase`
3. `cd /tmp/testcase`
4. `CCACHE=~/src/ccache/_build/ccache ~/src/ccache/test/run`

### Actual behavior ###

```
Test suite:     basedir (line 291)
Test case:      Absolute paths in stderr
Failure reason: reference.stderr and ccache.stderr differ
```

The diff is the following:

```diff
diff -u ./testdir/84366/run.real/reference.stderr ./testdir/84366/run.real/ccache.stderr   
--- ./testdir/84366/run.real/reference.stderr	2021-08-22 13:25:29.000000000 +0200
+++ ./testdir/84366/run.real/ccache.stderr	2021-08-22 13:25:29.000000000 +0200
@@ -1,8 +1,8 @@
-In file included from /tmp/testcase/testdir/84366/run.real/test.c:1:
-/tmp/testcase/testdir/84366/run.real/test.h:1:2: warning: test.h [-W#warnings]
+In file included from /private/tmp/testcase/testdir/84366/run.real/test.c:1:
+/private/tmp/testcase/testdir/84366/run.real/test.h:1:2: warning: test.h [-W#warnings]
 #warning test.h
  ^
-/tmp/testcase/testdir/84366/run.real/test.c:2:2: warning: test.c [-W#warnings]
+/private/tmp/testcase/testdir/84366/run.real/test.c:2:2: warning: test.c [-W#warnings]
 #warning test.c
  ^
 2 warnings generated.
```

That might happen because `/tmp` is actually a symlink to `private/tmp`.

Thanks,
Gregor

**Comment by jrosdahl** [comment]
I think this should be fixed by 2397a0605a24d61d60738fa21591510a9fa24eed. Please give it a try.

**Comment by gjasny** [comment]
Thanks for fixing. It works as expected. The ccache test suite now passes during homebrew builds: Homebrew/homebrew-core#83900.
